desktop-widgets/command_divesite.cpp

```diff
@@ -186,6 +186,18 @@ static void swap(char *&c, QString &q)
 	q = s;
 }
 
+// Helper function: collect the dives that are at the given dive site
+static QVector<dive *> getDivesForSite(struct dive_site *ds)
+{
+	QVector<dive *> diveSiteDives;
```

Comment:
Premature optimization, but why not:
```
	diveSiteDives.reserve(ds->dives.nr);
```
?

Reply:
agreed. I should use that.